driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/gridfs/GridFSUploadPublisherImpl.java

```diff
@@ -120,7 +120,7 @@ public void subscribe(final Subscriber<? super Void> s) {
                                         return originalError;
                                     })
                                     .then(Mono.error(originalError)))
-                    .doOnCancel(() -> createCancellationMono(terminated, timeout).subscribe())
+                    .doOnCancel(() -> createCancellationMono(terminated, timeout).contextWrite(ctx).subscribe())
```

Comment:
Cancellation has a side effect - so we need to propagate context there.

Reply:
Isn't us calling `subscribe()` here not with a subscriber that was given to us (the `s` argument), the only thing that requires us to propagate the context from the subscriber `s`? I suspect that the fact that `createCancellationMono` has a side-effect (it mutates `terminated`) is irrelevant. If the side-effect is relevant to context propagation, then could you please explain why?

Reply:
On cancellation of the main `Publisher` we are calling [subscribe()](https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Mono.html#subscribe--) on a async clean up `Publisher` and we ignore any errors, results / completion.

> If the side-effect is relevant to context propagation, then could you please explain why?

The user may have tracing and want to trace from the web app to the driver. Without adding the context here we break that chain and they cannot tie the cleanup operations to a web request.